drivers/hmis/app/graphql/types/hmis_schema/external_form_submission.rb

```diff
@@ -34,5 +37,34 @@ def values
       # the client name that was originally submitted with the form.
       object.form_values
     end
+
+    # "Summary" fields are a subset of the form's fields that are displayed on the external forms review table.
+    # Includes Client first/last name plus any CDEDs where show_in_summary is true.
+    def summary_fields
+      cdeds_by_key = load_ar_association(definition, :custom_data_element_definitions).
+        select(&:show_in_summary).
+        index_by(&:key).stringify_keys
+
+      cded_keys = cdeds_by_key.keys
+
+      object.form_values.map do |key, value|
+        case key.to_s
+        when 'Client.firstName'
+          OpenStruct.new(id: 'first', key: 'First Name', value: value)
+        when 'Client.lastName'
+          OpenStruct.new(id: 'last', key: 'Last Name', value: value)
+        when *cded_keys
+          cded = cdeds_by_key[key]
+          OpenStruct.new(id: cded.id, key: cded.label, value: value)
+        end
+      end.compact.sort_by(&:key)
+    end
```

Comment:
This is fine. There's probably so few cdeds here that it doesn't matter but some small optimizations could be made:
- a single struct definition should be more memory efficient than calling OpenStruct in a loop
- use a set for cded_keys should make the loop a little faster 

```suggestion
    Field = Struct.new(:id, :key, :value, keyword_init: true)
    def summary_fields
      cdeds_by_key = load_ar_association(definition, :custom_data_element_definitions).
        select(&:show_in_summary).
        index_by(&:key).stringify_keys

      cded_keys = cdeds_by_key.keys.to_set

      object.form_values.map do |key, value|
        case key.to_s
        when 'Client.firstName'
          Field.new(id: 'first', key: 'First Name', value: value)
        when 'Client.lastName'
          Field.new(id: 'last', key: 'Last Name', value: value)
        when cded_keys
          cded = cdeds_by_key[key]
          Field.new(id: cded.id, key: cded.label, value: value)
        end
      end.compact.sort_by(&:key)
    end
```

Reply:
Thanks that;'s good to know about Open Structs, updated to match your suggestion